tests/hpc/package_check.pm

```diff
@@ -0,0 +1,139 @@
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: HPC package(s) checks
+#  The test is meant as rudimentary check of some white-listed specific
+#  packages. The white list should be located in data folder and should
+#  contain the list of rpms with expected minimal versions. This white
+#  list is then used for basic checks if packages are installable, if
+#  the version is as expected and then single basic tests is run for
+#  each package.
+# Maintainer: Sebastian Chlad <sebastian.chlad@suse.com>
+
+use base 'hpcbase';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub check_version {
+    my ($rpm, $expected_version) = @_;
+
+    my $installed_rpm_ver;
+
+    if ($rpm =~ s/papi//) {
+        $installed_rpm_ver = script_output("rpm -qa | grep papi");
+        $installed_rpm_ver =~ s/papi-//;
+	$installed_rpm_ver =~ s/[.]//g;
+	$installed_rpm_ver = substr($installed_rpm_ver, 0, 3);
+    } else {
+        $installed_rpm_ver = script_output("$rpm -v");
+        $installed_rpm_ver =~ s/$rpm version//;
+        $installed_rpm_ver =~ s{^\s+|\s+$}{}g;
+    }
+
+    ## make casting explicit
+    my $num1 = int($expected_version);
+    my $num2 = int($installed_rpm_ver);
+
+    if ($num1 == $num2) {
+        record_info('PASS: ver. check', script_output("echo $rpm version is satisfied"));
+    } else {
+        record_info('FAIL: ver. check', script_output("echo $rpm version is not satisfied"));
+        record_info('DEBUG', script_output("echo comparison: $num1: $num2"));
+    }
+}
+
+sub run_test {
+    my ($rpm) = @_;
+
+    record_info("test: $rpm");
+    if ($rpm eq "cpuid") {
+        test_cpuid($rpm);
+    } elsif ($rpm eq "papi") {
+        test_papi($rpm);
+    }
+}
+
+sub test_cpuid {
+    my ($rpm) = @_;
+
+    ##TODO: this must be validated
+    assert_script_run("cpuid --one-cpu");
+    record_info("PASS: $rpm", script_output("echo $rpm basic cmd can be run"));
+}
+
+sub test_papi{
+    my ($rpm) = @_;
+
+    record_info("PASS: $rpm", script_output("echo $rpm basic cmd can be run"));
+}
+
+sub acquire_rpm_list {
+    my ($version) = @_;
+    my $rpm_file = "package_list-$version";
+    my @rpm_list;
+
+    assert_script_run("wget --quiet " . data_url("hpc/$rpm_file") . " -O /tmp/$rpm_file");
+    assert_script_run("wget --quiet " . data_url("hpc/pars_packages_list.pl") . " -O pars_packages_list-$version.pl");
+
+    assert_script_run("chmod +x pars_packages_list-$version.pl");
+    my $rpm_list = script_output("./pars_packages_list-$version.pl");
+
+    @rpm_list = split(/\|/, $rpm_list);
+
+    return @rpm_list;
+}
+
+sub run {
+    my $self    = shift;
+    my $version = get_required_var('VERSION');
+    my $arch    = get_required_var('ARCH');
+
+    ## Install expected, white-listed rpms
+    my @rpms;
+    @rpms = acquire_rpm_list($version);
```

Comment:
looks like you transfer a file to SUT, then analyse it there with perl and transfere the result back to worker (isotovideo).
Why not just access the data dir directly like
```
my $fname = get_required_var('CASEDIR') . "/data/hpc/package_list-$version";
```
```